doc/endpoint/pickler.md

```diff
@@ -87,9 +87,10 @@ import sttp.tapir.generic.Configuration
 given customConfiguration: Configuration = Configuration.default.withSnakeCaseMemberNames
 ```
 
-## Enums / sealed traits / coproducts
+## Sealed traits / coproducts
 
-Pickler derivation for coproduct types (enums / sealed hierarchies) works automatically, by adding an `$type` discriminator field with the full class name. This is the default behavior of uPickle, but it can be overridden either by changing the discriminator field name, or by using custom logic to get field value from base trait.
+Pickler derivation for coproduct types (enums with parameters / sealed hierarchies) works automatically, by adding a `$type` discriminator field with the full class name. This is the default behavior of uPickle, but it can be overridden either by changing the discriminator field name, or by using custom logic to get field value from base trait.
```

Comment:
Hm will that match the generated docs - that is, do we provide a pickler-specific `Configuration` which matches this behavior?

Reply:
Good catch! This would almost certainly diverge in the schema, I'll add some tests and fixes accordingly.

Reply:
I think that pickler has to provide its own configuration - setting the discriminator field to `$type`. Question then is, how to sensibly provide such an implicit which has higher priority than the default one (should be easy, I think almost everything has higher priority than companion object implicits), but also in a way that's overridable by the user with a custom configuration. Can this be done at all?

As for whether to use short or long names - that's a good question ;) I think we should pick a sensible default, not necessarily follow current tapir's or upickle defaults. Full names seem to expose too much implementation details (the package name) to me, so I think short names are the better option. As for the discriminator field, `$type` seems to be as good as any.